sbi/inference/posteriors/base_posterior.py

```diff
@@ -697,23 +702,45 @@ def sample_conditional(
 
         x, num_samples = self._prepare_for_sample(x, sample_shape)
 
-        cond_potential_fn_provider = ConditionalPotentialFunctionProvider(
-            potential_fn_provider, condition, dims_to_sample
-        )
-
         if sample_with == "mcmc":
             mcmc_method, mcmc_parameters = self._potentially_replace_mcmc_parameters(
                 mcmc_method, mcmc_parameters
             )
-            samples = self._sample_posterior_mcmc(
+            transform = mcmc_transform(
+                self._prior, device=self._device, **mcmc_parameters
+            )
+            transform_dims_to_sample = RestrictedTransformForConditional(
+                transform, condition, dims_to_sample
+            )
+
+            condition = atleast_2d_float32_tensor(condition)
+
+            # Transform the `condition` to unconstrained space.
+            tf_condition = transform.inv(condition)
```

Comment:
overall I would vote for writing out `tf_` to `transformed_` if space allows, but feel free to leave it like it is now.